### ISSUE TITLE
[Dependencies] Global 

### ISSUE THREAD

**Issue by romainseb** [Talend/ui#2755]
Here are listed only the major upgrade for dependencies, there is a lot of minor once but a yarn upgrade should be enough for them.
The dependencies list there would appear one time by order of appearance.
For example, redux-saga is in several packages but it's list only once in the first package encountered

## @talend/react-cmf

- [] Use TUI-script to build it

### DEPS

| Name                    | Current | Latest | Description                              | Action                                               | Done |
| ----------------------- | ------- | ------ | ---------------------------------------- | ---------------------------------------------------- | ---- |
| deepmerge               | 1.5.1   | 4.2.2  | used to merge the settings               | Upgrade                                              | []   |
| hoist-non-react-statics | 2.5.5   | 3.3.2  | used in cmfConnect                       | Why do we need it ?                                  | []   |
| immutable               |         |        |                                          | Analyse the impact of a removal                      | []   |
| mkdirp                  | 0.5.1   | 1.0.4  | utilisé dans des scripts de merge        | Upgrade                                              | []   |
| path-to-regexp          | 2.0.0   | 6.1.0  | used in matchPath forked from router v4  | Dangerous upgrade                                    | []   |
| react-redux             | 5.0.7   | 7.2.0  | Compatible with redux 3 / hooks          | POC move to peerDep and let the app choose a version | []   |
| redux                   | 3.7.2   | 4.0.5  |                                          | POC to make                                          | []   |
| redux-batched-actions   | 0.2.0   | 0.5.0  |                                          | Upgrade                                              | []   |
| redux-saga              | 0.15.6  | 1.1.3  | some function have moved, not a big deal | Upgrade                                              | []   |

https://github.com/reduxjs/react-redux/blob/master/package.json#L40
https://github.com/reduxjs/redux/issues/1342

### DEV DEPS

The best way to fix this deps would be to move on TUI-Script to build the library

| Name                   | Current | Latest | Description                           | Action  | Done |
| ---------------------- | ------- | ------ | ------------------------------------- | ------- | ---- |
| babel-jest             | 24.7.1  | 25.3.0 |                                       |         | []   |
| eslint                 | 4.0.0   | 6.8.0  |                                       |         | []   |
| eslint-config-airbnb   | 11.1.0  | 18.1.0 |                                       |         | []   |
| eslint-plugin-import   | 1.16.0  | 2.20.2 |                                       |         | []   |
| eslint-plugin-jsx-a11y | 2.2.2   | 6.2.3  | not sure it's used in the cmf package |         | []   |
| jest                   | 24.7.1  | 25.3.0 |                                       |         | []   |
| node-fetch             | 1.7.3   | 2.6.0  | for some mocks                        | Upgrade | []   |

## @talend/react-cmf-cqrs

- [] Use TUI-script to build it

### DEV DEPS

The best way to fix this deps would be to move on TUI-Script to build the library

| Name                | Current | Latest | Description    | Action                                                 | Done |
| ------------------- | ------- | ------ | -------------- | ------------------------------------------------------ | ---- |
| eslint-plugin-react | 6.3.0   | 7.19.0 |                | Upgrade                                                | []   |
| mock-socket         | 7.0.0   | 9.0.3  | for some tests | Y                                                      | []   |
| rimraf              | 2.6.2   | 3.0.2  | rm -rf ?       | Make a poll to drop native windows usage as WSL exists | []   |

## @talend/react-cmf-webpack-plugin

- [] Use TUI-script to build it

## DEPS

| Name      | Current | Latest | Description | Action  | Done |
| --------- | ------- | ------ | ----------- | ------- | ---- |
| chokidar  | 2.0.2   | 3.3.1  |             | Upgrade | []   |
| deepmerge | 1.5.1   | 4.2.2  |             | Upgrade | []   |
| mkdirp    | 0.5.1   | 1.0.4  |             | Upgrade | []   |

## @talend/react-components

- [] Use TUI-script to build it
- [] Analysis on the creation of a storybook-preset in order to not have some dev dep there to run the SB

### DEPS

| Name                    | Current | Latest | Description                                        | Action               | Done |
| ----------------------- | ------- | ------ | -------------------------------------------------- | -------------------- | ---- |
| date-fns                | 1.27.2  | 2.12.0 |                                                    | Upgrade              | []   |
| rc-slider               | 8.6.3   | 9.2.4  |                                                    | Upgrade              | []   |
| rc-tooltip              | 3.7.3   | 4.0.3  |                                                    | Upgrade              | []   |
| react-bootstrap         | 0.32.4  | 1.0.2  | Huge impacts                                       | Won't do             | []   |
| react-dnd               | 2.5.4   | 10.0.2 | The component that use it is not used              | Remove               | []   |
| react-dnd-html5-backend | 2.5.4   | 10.0.2 | The component that use it is not used              | Remove               | []   |
| react-draggable         | 3.3.2   | 4.3.1  |                                                    | Upgrade              | []   |
| react-popper            | 1.3.3   | 2.2.2  |                                                    | Upgrade              | []   |
| react-transition-group  | 2.3.1   | 4.3.0  | See to remove the deprecated one (used in drawers) | Upgrade              | []   |
| react-use               | 13.27.0 | 14.1.1 |                                                    | Upgrade              | []   |
| simplebar               | 4.2.2   | 5.1.0  | Nothing breaking on react side afaik               | Upgrade              | []   |
| simplebar-react         | 1.2.2   | 2.1.0  | Nothing breaking on react side afaik               | Upgrade              | []   |
| styled-components       | 4.1.3   | 5.1.0  | Look at reactour deps (as it's used inside)        | Analysis and Upgrade | []   |
| timezone-support        | 1.5.5   | 2.0.2  |                                                    | Upgrade              | []   |
| uuid                    | 3.0.1   | 7.0.3  |                                                    | Upgrade              | []   |

### PEER DEPS

| Name          | Current | Latest | Description             | Action  | Done |
| ------------- | ------- | ------ | ----------------------- | ------- | ---- |
| i18next       | 15.1.3  | 19.4.2 | Take care of extraction | Upgrade | []   |
| react-i18next | 10.10.0 | 11.3.4 | Take care of extraction | Upgrade | []   |

### DEV DEPS

| Name         | Current | Latest | Description        | Action            | Done |
| ------------ | ------- | ------ | ------------------ | ----------------- | ---- |
| autoprefixer | 7.1.4   | 9.7.6  | Linked to SB       | Upgrade           | []   |
| cross-env    | 5.2.0   | 7.0.2  |                    | Upgrade           | []   |
| css-loader   | 1.0.1   | 3.5.2  | Linked to SB       | Upgrade           | []   |
| jsdom        | 11.11.0 | 16.2.2 | Used in some tests | Upgrade           | []   |
| react-a11y   | 0.3.3   | 1.1.0  | Deprecated         | move to react-axe | []   |
| sass-loader  | 7.1.0   | 8.0.2  | Linked to SB       | Upgrade           | []   |
| slimerjs     | 0.10.2  | 1.0.0  |                    | to Remove         | []   |
| style-loader | 0.23.0  | 1.1.4  | Linked to SB       | Upgrade           | []   |

## @talend/react-containers

- [] Use TUI-script to build it

| Name         | Current | Latest | Description         | Action                                                            | Done |
| ------------ | ------- | ------ | ------------------- | ----------------------------------------------------------------- | ---- |
| memoize-one  | 4.0.0   | 5.1.1  | useMemo or reselect | used in class components, need to migrate them first to remove it | []   |
| reselect     | 2.5.4   | 4.0.0  |                     | Upgrade                                                           | []   |
| react-router | 3.2.0   | 5.1.2  | used in sidepanel   | Analysis                                                          | []   |

## @talend/react-datagrid

- [] Use TUI-script to build it

| Name              | Current | Latest | Description | Action  | Done |
| ----------------- | ------- | ------ | ----------- | ------- | ---- |
| ag-grid-community | 21.2.1  | 23.0.2 |             | Upgrade | []   |
| ag-grid-react     | 21.2.1  | 23.0.3 |             | Upgrade | []   |
| @talend/react-cmf | 1.6.0   | 5.0.0  | 😓          | Upgrade | []   |

## @talend/react-forms

- [] Use TUI-script to build it

| Name                  | Current | Latest | Description            | Action   | Done |
| --------------------- | ------- | ------ | ---------------------- | -------- | ---- |
| react-jsonschema-form | 0.51.0  | 1.8.1  | used in TCOMP v0 / TDP | Won't do | [x]  |
| react-ace             | 6.2.0   | 8.1.0  |                        | Upgrade  | []   |

## generator-talend

As it's really a deprecated generator, stuff evolved fast and it's not maintained, it would be better to remove it as it's not used.

| Name             | Current | Latest | Description | Action | Done |
| ---------------- | ------- | ------ | ----------- | ------ | ---- |
| chalk            | 1.1.3   | 4.0.0  |             | Remove | []   |
| yeoman-generator | 3.1.1   | 4.8.2  |             | Remove | []   |
| yosay            | 1.0.0   | 2.0.2  |             | Remove | []   |
| mocha            | 5.2.0   | 7.1.1  |             | Remove | []   |
| yeoman-test      | 1.9.1   | 2.4.1  |             | Remove | []   |

## @talend/html-webpack-plugin

| Name                     | Current | Latest | Description | Action                                  | Done |
| ------------------------ | ------- | ------ | ----------- | --------------------------------------- | ---- |
| @talend/react-components | 2.3.0   | 5.0.1  | 😓          | Upgrade and add it in the lerna package | []   |

## Icons

- [] Use TUI-script to build it

## talend/json-schema-form-core

Forked, will stay this way for now

## @talend/local-libs-webpack-plugin

Currently used only in TPD, see if it's still useful to keep it

## @talend/react-cmf-router

- [] Use TUI-script to build it

Nothing more as above

## sagas

- [] Use TUI-script to build it

## scripts

Was the old version of TUI-Script

- [] Removal

## stepper

- [] Use TUI-script to build it
